### ISSUE TITLE
Support github apikey without requiring environment variables

### ISSUE THREAD

**Issue by eli-schwartz** [lilydjwg/nvchecker#69]
I'd much prefer to use another __config__ option, e.g. `apikey = ~/.config/nvchecker/apikey`, where this file contains the OAuth token.

More generally this could probably collect the various gitlab tokens, and I think this is everything which uses tokens.

**Comment by lilydjwg** [comment]
I've implemented such a feature in the [keyfile branch](https://github.com/lilydjwg/nvchecker/tree/keyfile). Give it a try?

**Comment by eli-schwartz** [comment]
Installed it to test, and I hit two snags:

I added a keyfile, then tried running nvchecker, and got:

`/bin/bash: /home/eschwartz/.local/lib/python3.6/site-packages/nvchecker-1.1.dev0-py3.6.egg/nvchecker/source/vcs.sh: Not a directory`

Because naturally `python setup.py install --user` defaults to installing a zipped egg, and the setup declares zip_safe = True (even though it is not at all safe).

So, modifying that, things seem good (and my VCS source does not spawn errors).

But as a last test, I tried removing the keyfile option, and got lots of:
```
[E 05-05 21:41:41.426 _base:136] sigil: unexpected error happened
    Traceback (most recent call last):
      File "/usr/lib/python3.6/configparser.py", line 1138, in _unify_values
        sectiondict = self._sections[section]
    KeyError: 'keys'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/home/eschwartz/.local/lib/python3.6/site-packages/nvchecker-1.1.dev0-py3.6.egg/nvchecker/core.py", line 148, in worker
        ret = await get_version(name, conf, keyman=self.keymanager)
      File "/home/eschwartz/.local/lib/python3.6/site-packages/nvchecker-1.1.dev0-py3.6.egg/nvchecker/get_version.py", line 45, in get_version
        version = await func(name, conf, **kwargs)
      File "/home/eschwartz/.local/lib/python3.6/site-packages/nvchecker-1.1.dev0-py3.6.egg/nvchecker/source/github.py", line 42, in get_version
        key = kwargs['keyman'].get_key('github')
      File "/home/eschwartz/.local/lib/python3.6/site-packages/nvchecker-1.1.dev0-py3.6.egg/nvchecker/core.py", line 204, in get_key
        return self.config.get('keys', name)
      File "/usr/lib/python3.6/configparser.py", line 781, in get
        d = self._unify_values(section, vars)
      File "/usr/lib/python3.6/configparser.py", line 1141, in _unify_values
        raise NoSectionError(section)
    configparser.NoSectionError: No section: 'keys'
```

Though, it is definitely using the key when properly available, since if I use a fake OAuth token I get 401 unauthorized errors. :smile:

**Comment by lilydjwg** [comment]
Those issues should have been fixed.

**Comment by eli-schwartz** [comment]
Not quite...

`configparser.NoOptionError: No option 'github' in section: 'keys'`

I'd suggest catching missing keyfiles (try/except) and ignoring the error (pass), since it should not be an error for github, and for gitlab it already uses the custom error: https://github.com/lilydjwg/nvchecker/blob/6bcef4e30700b7b1b7c7f537577608843d54391e/nvchecker/source/gitlab.py#L36

**Comment by eli-schwartz** [comment]
Probably best to make get_key return a blank string (or NoneType) if the key doesn't exist. It's a dict lookup, so you could just provide a fallback value without messing with try/except.

**Comment by eli-schwartz** [comment]
Yes, I did a ninja edit. :D

Pulled the new commit, reinstalled nvchecker, and it works. Removing the keyfile option simply means it works until after a couple runs when I start getting "403 forbidden" due to API overuse, using the keyfile allows me to get updates properly.

Thanks.

(During the process of testing, it came to my mind that perhaps 403 forbidden errors should be trapped and replaced with a single log error stating that Github's API was overused and using a token could help. It certainly confused me the first time I got the errors.)
